questionnaire_linking.py

```diff
@@ -0,0 +1,117 @@
+import csv
+import argparse
+import requests
+import qid_checksum_validator
+from config import Config
+from datetime import datetime
+import uuid
+import json
+from utilities.rabbit_context import RabbitContext
+
+
+def validate_qid_link_file(qid_link_file_path):
+    try:
+        with open(qid_link_file_path, encoding="utf-8") as qid_link_file:
+            return qid_file(qid_link_file)
+    except UnicodeDecodeError as err:
+        print(f'Invalid file encoding, requires utf-8, error: {err}')
+        exit(-1)
+
+
+def qid_file(qid_link_file):
+    qid_link_file_reader = csv.DictReader(qid_link_file, delimiter=',', fieldnames=['case_ref', 'qid'])
+    validate_and_submit_questionnaire_links(qid_link_file_reader)
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description='Load a QID Link file into response management.')
+    parser.add_argument('qid_link_file_path', help='path to the QID Link file', type=str)
+    return parser.parse_args()
+
+
+args = parse_arguments()
+error_count = 0
+
+
+def validate_and_submit_questionnaire_links(qid_link_file_reader):
+    global error_count
+    for line_number, questionnaire_link in enumerate(qid_link_file_reader, 1):
+        error_count = 0
+        for i in questionnaire_link:
+            questionnaire_link[i] = questionnaire_link[i].replace(" ", "")
+        validate_qid_len_and_type(line_number, questionnaire_link["qid"])
+        validate_qid_third_digit(line_number, questionnaire_link["qid"])
+        validate_check_digits(line_number, questionnaire_link["qid"])
+        case_id = validate_case_ref(line_number, questionnaire_link["case_ref"])
+
+        if error_count > 0:
+            print(f'Error: {questionnaire_link} on line {line_number} have FAILED')
```

Comment:
I don't think you need to use a global or count the errors here, the validate functions could return booleans, or the error count could be local and passed in.
Bit nitpicky but it's best to avoid unnecessary use of globals/side effect based functions.